src/utilities/NameRandomizer.py

```diff
@@ -0,0 +1,79 @@
+#Creates a randomized name from a dictionary of first and last names. Created
+#as a non-class utility due to expected low request volume and to allow for
+#implementation changes without affecting the parent.
+#
+#A dictionary-style format is used for user convenience and compatibility.
+
+
+#####  Imports  #####
+
+import linecache as lc
+import logging as log
+import os
+import random as rand
+
+#####  Package Variables  #####
+
+fn_dict_path = None
+fn_dict_size = 0
+ln_dict_path = None
+ln_dict_size = 0
+name_log     = None
+
+#####  Package Functions  #####
+
+def init(options : dict):
+    """Saves relevant settings to avoid repetitive reads to dictionaries.
+       It gets all relevant parameters from the parent and assumes they
+       have already been sanitized.
+
+       Input: self - Pointer to the current object instance.
+              options - a dict of all necessary dictionary paths and sizes.
+
+       Output: None - Throws exceptions on error.
+    """
+    global name_log
+    global fn_dict_path
+    global fn_dict_size
+    global ln_dict_path
+    global ln_dict_size
+
+    #Maybe this should be a separate log at some point.
+    name_log  = log.getLogger('discord')
+    #This is deliberately not a Path, like in the parent, to allow linecache to
+    #read the file in case the user decided to provide a large file.
+    #Also, apparently Pathlib is missing methods that cachelib is using
+    #(probably for sanitization), causing exceptions if you give it a Path.
+    fn_dict_path = os.path.abspath(options['fn_dict_path'])
+    fn_dict_size = int(options['fn_dict_size'])
+    ln_dict_path = os.path.abspath(options['ln_dict_path'])
+    ln_dict_size = int(options['ln_dict_size'])
+
+    #TODO: add a RNG initializer if this is ever split into its own thread.
+
+    name_log.info(f"Initializing the Name RNG class with: first name dict path: {fn_dict_path} fn_dict_size: {fn_dict_size} last name dict path: {ln_dict_path} ln_dict_size: {ln_dict_size}.")
+
+def GetRandomName() -> str:
+    """Creates a randomized name from the name dictionaries provided during
+       initialization.
+
+       Input: self - Pointer to the current object instance.
+
+       Output: str - A space-separated full name in Western order.
+    """
+    global name_log
+    global fn_dict_path
+    global fn_dict_size
+    global ln_dict_path
+    global ln_dict_size
+    first_name = "Default"
+    last_name  = "Sally"
+
+    #TODO: Consider adding the ability for a variable length name string return.
+
+    first_name = (lc.getline(fn_dict_path, rand.randint(0, fn_dict_size))).rstrip()
```

Comment:
Are names going to be unique even when the system gets reset? Maybe should query the database to get a list of names already used?

Reply:
Names are not unique, I also didn't want to impose that on someone else running the code and using a different name DB.  It's also why the system relies on MariaDB's UUIDs to actually track profiles/pictures.

The current name DB generates something on the order of 61 million unique names, but they're only first name + last name combos.  It would be nice to allow for some flexibility in the name format used (just Mononyms, maybe a full 4+ word name), but it's not required for the initial version.